pkgs/by-name/ec/ecapture/package.nix

```diff
@@ -0,0 +1,129 @@
+{
+  buildGoModule,
+  fetchFromGitHub,
+  stdenv,
+  lib,
+  nspr,
+  libpcap,
+  clang,
+  fd,
+  go-bindata,
+  glibc,
+  gnutls,
+  bash,
+  postgresql,
+  mariadb,
+  openssl,
+}:
+
+buildGoModule rec {
+  pname = "ecapture";
+  version = "0.8.6";
+
+  src = fetchFromGitHub {
+    owner = "gojue";
+    repo = "ecapture";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-xnUgsnz3zUkuLwqgdogEWQh0GMEmS/qmDqqmEQlHhfQ=";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [
+    clang
+    fd
+    go-bindata
+  ];
+
+  newlibpcap = libpcap.overrideAttrs (previousAttrs: {
+    configureFlags = previousAttrs.configureFlags ++ [ "--without-libnl" ];
+  });
+
+  buildInputs = [
+    newlibpcap
+    glibc.static
+    glibc
+  ];
+
+  ldflags = [
+    "-extldflags -static"
+  ];
+
+  CGO_LDFLAGS = "-lpcap -static";
+
+  doCheck = false;
+
+  hardeningDisable = [
+    "zerocallusedregs"
+  ];
+
+  patchPhase = ''
+    runHook preHook
+
+    substituteInPlace user/config/config_gnutls_linux.go \
+      --replace-fail 'return errors.New("cant found Gnutls so load path")' 'gc.Gnutls = "${lib.getLib gnutls}/lib/libgnutls.so.30"' \
+      --replace-fail '"errors"' ' '
+
+    substituteInPlace user/module/probe_bash.go \
+      --replace-fail '/bin/bash' '/run/current-system/sw/bin/bash'
+
+    substituteInPlace user/config/config_bash.go \
+      --replace-fail '/bin/bash' '/run/current-system/sw/bin/bash'
+
+    substituteInPlace user/config/config_nspr_linux.go \
+      --replace-fail '/usr/lib/firefox/libnspr4.so' '${lib.getLib nspr}/lib/libnspr4.so'
+
+    substituteInPlace cli/cmd/postgres.go \
+      --replace-fail '/usr/bin/postgres' '${postgresql}/bin/postgres'
+
+    substituteInPlace cli/cmd/mysqld.go \
+      --replace-fail '/usr/sbin/mariadbd' '${mariadb}/bin/mariadbd'
+
+    substituteInPlace user/module/probe_mysqld.go \
+      --replace-fail '/usr/sbin/mariadbd' '${mariadb}/bin/mariadbd'
+
+    substituteInPlace user/config/config_openssl_linux.go \
+      --replace-fail 'return errors.New("cant found openssl so load path")' 'oc.Openssl = "${lib.getLib openssl}/lib/libssl.so.3"' \
+      --replace-fail '"errors"' ' '
+
+    runHook postHook
+  '';
+
+  postConfigure =
+    (
+      if stdenv.isx86_64 then
+        ''
+          fd "kern/*.c" --strip-cwd-prefix -x clang -D__TARGET_ARCH_x86 \
+             -O2 -mcpu=v1  -nostdinc -Wno-pointer-sign \
+             -I ./kern -I ./kern/bpf/x86 \
+             -target bpfel -c {} -o ./user/bytecode/{/.}_core.o \
+             -fno-ident -fdebug-compilation-dir . -g \
+             -MD -MP
+        ''
+      else
+        ''
+          fd "kern/*.c" --strip-cwd-prefix -x clang -D__TARGET_ARCH_arm64 \
+             -O2 -mcpu=v1  -nostdinc -Wno-pointer-sign \
+             -I ./kern -I ./kern/bpf/arm64 \
+             -target bpfel -c {} -o ./user/bytecode/{/.}_core.o \
+             -fno-ident -fdebug-compilation-dir . -g \
+             -MD -MP
+        ''
+    )
```

Comment:
Not really sure about this part. Also is it certain that we can't get something that's architecture agnostic?